poutyne/framework/experiment.py

```diff
@@ -331,24 +518,58 @@ def load_best_checkpoint(self, *, verbose=False):
         return best_epoch_stats
 
     def load_checkpoint(self, epoch):
+        """
+        Loads the attribute model's weights with the weights at a given checkpoint epoch.
+
+        Args:
+            epoch (int): The checkpoint epoch to load.
+        """
         ckpt_filename = self.best_checkpoint_filename.format(epoch=epoch)
         self.model.load_weights(ckpt_filename)
 
     def load_last_checkpoint(self):
+        """
+        Loads the attribute model's weights with the weights of the last checkpoint.
+        """
         self.model.load_weights(self.model_checkpoint_filename)
 
-    def test(self, test_loader, *, steps=None, load_best_checkpoint=True, load_last_checkpoint=False, seed=42):
-        if seed is not None:
-            # Make training deterministic.
-            random.seed(seed)
-            np.random.seed(seed)
-            torch.manual_seed(seed)
+    def test(self, test_loader, *, steps=None, checkpoint='best', seed=42):
+        """
+        Computes and returns the loss and the metrics of the attribute model on a given test examples
+        loader.
+
+        Args:
+            test_loader: Generator-like object for the test set. See :func:`~Model.fit_generator()` for
+                details on the types of generators supported.
+            steps (int, optional): Number of iterations done on ``generator``.
+                (Defaults the number of steps needed to see the entire dataset)
+            checkpoint (Union[str, int]): Which model checkpoint weights to load for the test evaluation.
+                If 'best', will load the best weights according to ``monitor_metric`` and ``monitor_mode``.
+                If 'last', will load the last model checkpoint. If int, will load the checkpoint of the
+                specified epoch.
+                (Default value = 'best')
+            load_best_checkpoint (bool, optional): Whether or not to load the best checkpoint's weights.
+                If set to true, the ``load_last_checkpoint`` argument is ignored.
+                (Default value = True)
+            load_last_checkpoint (bool, optional): Whether or not to load the last checkpoint's weights.
+                (Default value = False)
+            seed (int, optional): Seed used to make the sampling deterministic.
+                (Default value = 42)
+
+        Returns:
+            dict sorting of all the test metrics values by their names.
+        """
+        set_seeds(seed)
 
         best_epoch_stats = None
-        if load_best_checkpoint:
+        if checkpoint == 'best':
```

Comment:
I think it would be better to move this logic into a private method (something like `self._resolve(checkpoint)` which will load the appropriated checkpoint.
Reason: i'm not quite sure it's the `test` method to resolve with checkpoint it should use. 
From the doc : Computes and returns the loss and the metrics of the attribute model on a given test examples loader.

Reply:
maybe load_checkpoint should do the job

Reply:
Agreed. I moved all the logic in self.load_checkpoint() and prepended _ to the load_best, load_last and load_epoch methods in e06b62588b6dab600542d8d19884373514f28231